tests/rustdoc-json/structs/field_order.rs

```diff
@@ -0,0 +1,36 @@
+// Check that the order of fields is preserved.
+
+pub struct Foo {
+    pub foo_0: i32,
+    pub foo_1: i32,
+    pub foo_2: i32,
+    pub foo_3: i32,
+    pub foo_4: i32,
+    pub foo_5: i32,
+    pub foo_6: i32,
+    pub foo_7: i32,
+    pub foo_8: i32,
+    pub foo_9: i32,
```

Comment:
Might it be worth also trying to check whether field ordering is preserved even when the field names aren't lexicographically ordered when read in definition order?

Right now, this test can't distinguish between "field ordering is preserved" and "fields are always shown in lexicographic order." And since `BTreeMap` is very tempting to use in JSON for determinism reasons, "always in lexicographic order" is a very plausible implementation detail.

Reply:
Good catch, I've replace `foo` with 3 random characters.